Library/Homebrew/language/node.rb

```diff
@@ -41,14 +41,18 @@ def self.std_npm_install_args(libexec)
       pack = pack_for_installation
 
       # npm install args for global style module format installed into libexec
-      %W[
+      args = %W[
         -ddd
         --global
         --build-from-source
         --#{npm_cache_config}
         --prefix=#{libexec}
         #{Dir.pwd}/#{pack}
       ]
+
+      args << "--unsafe-perm" if ENV["USER"] == "root"
```

Comment:
Sorry to keep adding stuff but: could this use a Ruby class method like one on https://ruby-doc.org/core-2.5.1/Process/UID.html or https://ruby-doc.org/stdlib-2.0.0/libdoc/etc/rdoc/Etc.html? `id -u` has a process call overhead and `ENV["USER"]` can be overridden by the user to be wrong in general.

Reply:
`Process.uid` it is.